bluesky/plans.py

```diff
@@ -1631,10 +1629,19 @@ def list_scan(detectors, motor, steps, *, per_step=None, md=None):
         md,
         {'detectors': [det.name for det in detectors],
          'motors': [motor.name],
+         'num_steps': len(steps),
          'plan_args': {'detectors': list(map(repr, detectors)),
                        'motor': repr(motor), 'steps': steps,
                        'per_step': repr(per_step)},
-                       'plan_name': 'list_scan'})
+         'plan_name': 'list_scan',
+
+         # TODO what to do in this case?
+         'plan_pattern': 'list',
+         'plan_pattern_module': '__builtin__',
+         }
+    )
+
+    md['plan_pattern_args'] = [steps]
```

Comment:
Let's go with kwargs only (labels are good...) and side-step the problem with `list` by using `np.array` instead:

```python
In [4]: np.array(object=[1,2,3])
Out[4]: array([1, 2, 3])
```

so we can do `md['plan_pattern_args'] = dict(object=steps)`